<?xml version="1.0"?>
<launch>
  <arg name="use_dji" default="False" />
  <arg name="uav_odom_topic_name" default="/modified_odom" />
  <arg name="task_kind" default="1" />
  <arg name="rqt_gui" default="False" />
  <arg name="circle_motion_times" default="1" />
  <arg name="motion_planning" default="False" />

  ##### Perception Node #####
  <include file="$(find jsk_uav_forest_perception)/launch/tree_detector.launch">
    <arg name="simulation" value="True" unless="$(arg use_dji)" />
    <arg name="uav_odom_topic_name" value="$(arg uav_odom_topic_name)" />
  </include>

  ##### Motion Node #####
  <node pkg="jsk_uav_forest_motion" type="circle_motion.py" name="circle_motion" output="screen" args="$(arg use_dji)">
    <param name="task_kind" value="$(arg task_kind)" />
    <param name="motion_planning" value="$(arg motion_planning)" />
    <param name="circle_motion_times" value="$(arg circle_motion_times)" />
    <param name="uav_odom_sub_topic_name" value="$(arg uav_odom_topic_name)" if="$(arg use_dji)"/>
    <rosparam file="$(find jsk_uav_forest_motion)/config/circle_motion/circle_motion.yaml" command="load"/>
  </node>

  <group if="$(arg motion_planning)">
    <node pkg="jsk_uav_forest_motion" type="motion_plannar_client.py" name="motion_plannar" output="screen" args="$(arg use_dji)">
      <param name="task_kind" value="$(arg task_kind)" />
      <param name="uav_odom_sub_topic_name" value="$(arg uav_odom_topic_name)" if="$(arg use_dji)"/>
      <rosparam file="$(find jsk_uav_forest_motion)/config/circle_motion/circle_motion.yaml" command="load"/>
    </node>
  </group>
 
  ##### rqt gui #####
  <node pkg="rqt_gui" type="rqt_gui" name="rqt_gui" if="$(arg rqt_gui)" args="--perspective-file $(find jsk_uav_forest_motion)/launch/jsk_uav_forest.perspective"/>

</launch>
